tests/src/core/testqgslegendrenderer.cpp

```diff
@@ -1544,6 +1546,42 @@ void TestQgsLegendRenderer::testBigMarkerJson()
   QVERIFY( _verifyImage( test_name, mReport, 50 ) );
 }
 
+void TestQgsLegendRenderer::testLabelLegendJson()
```

Comment:
Can you add a rendering based test too please?

Reply:
Added a test against a reference image. When it comes to rendering based tests, I'm always a bit concerned about tests behaving differently depending on plattform (e.g. TestQgsLegendRenderer::testBasicJson fails on my developer machine).

Reply:
The best way to avoid that is to make the fonts bold and very large. (Ie I'd recommend bumping the font size in this new test). Then you can use the masking approach described here https://docs.qgis.org/3.16/en/docs/developers_guide/unittesting.html#comparing-images-for-rendering-tests to add tolerance to the test.

Reply:
Will look at it on Monday.
The test is indeed failing in the CI now (difference is very hard to detect by eye) and needs masking:

![test_label_legend](https://user-images.githubusercontent.com/278939/118228159-03f7f380-b48a-11eb-8543-561770798264.png)

Reply:
Ok, I'll approve and mark as freeze exempt!